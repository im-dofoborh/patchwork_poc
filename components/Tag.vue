<template>
  <router-link
    :to="route"
    class="standard-tag standard-tag--link" >
    {{ tag }}
  </router-link>
</template>

<script>
  export default {
    props: {
      tag: {
        type: String,
        required: true,
      }
    },

    computed: {
      route() {
        return {
          path: '/components/list',
          query: {
            search: this.tag.toLowerCase(),
          },
        }
      }
    }
  }
</script>
