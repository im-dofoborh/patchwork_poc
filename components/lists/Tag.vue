<template>
  <div class="sg-tag-list">
    <h3 class="sg-tag-list__title">Tags</h3>
    <ul class="sg-tag-list__list">
      <li>
        <router-link
          v-for="(tag, i) in routedTags"
          :key="i"
          :to="tag.route">
          <span class="sg-tag-list__item">
            {{ tag.name }}
          </span>
          <span
            v-if="i < routedTags.length - 1"
            class="sg-tag-list__seperator body-copy-small">
            |
          </span>
        </router-link>
      </li>
    </ul>
  </div>
</template>

<script>
  export default {
    props: {
      tags: {
        type: Array,
        default: () => [],
      }
    },

    computed: {
      routedTags() {
        return this.tags.map((tag) => {
          return {
            name: tag,
            route: {
              path: '/components/list',
              query: {
                search: tag.toLowerCase(),
              },
            }
          }
        })
      }
    }
  }
</script>

<style lang="scss" scoped>
  .sg-tag-list {
    display: flex;
    align-items: center;
    margin: spacing-unit(xxs) 0;

    &__title {
      font-size: 0.85em;
      color: $color--background--dark;
      font-weight: normal;
      margin-bottom: 0;
      margin-right: 10px;
    }

    &__list {
      display: flex;
      flex-wrap: wrap;
      margin: 0;
      padding: 0;
      list-style: none;
    }

    &__item {
      font-size: 0.85em;
      color: $color--clickable--regular;
      flex-shrink: 0;
    }

    &__seperator {
      padding: 0 spacing-unit(xxs);
    }
  }
</style>
