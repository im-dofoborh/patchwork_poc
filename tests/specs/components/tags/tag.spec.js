import { mount, createLocalVue } from '@vue/test-utils';
import VueRouter from 'vue-router';
import Tag from '~/components/Tag';

describe('Tag component', () => {
  let router = null;
  let localVue = null;
  let wrapper = null;

  beforeEach(() => {
    localVue = createLocalVue();
    localVue.component('tag', Tag);
    localVue.use(VueRouter);
    router = new VueRouter();
    wrapper = mount(Tag, {
      localVue,
      router,
      propsData: {
        tag: 'Foo',
      }
    })
  })

  it('renders default variant', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});
