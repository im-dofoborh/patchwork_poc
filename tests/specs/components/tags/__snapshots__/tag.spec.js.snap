// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tag component renders default variant 1`] = `
<a href="#/components/list?search=foo" class="standard-tag standard-tag--link">
  Foo
</a>
`;
